identity_did/src/did_url.rs

```diff
@@ -535,6 +534,33 @@ pub(crate) const fn is_char_fragment(ch: char) -> bool {
   is_char_path(ch) || ch == '?'
 }
 
+pub(crate) fn is_valid_percent_encoded_char(s: &str) -> bool {
+  let mut chars = s.chars();
+  let Some('%') = chars.next() else { return false };
+  s.len() == 3 && chars.take(2).all(|c| c.is_ascii_hexdigit())
```

Comment:
Maybe

```suggestion
  s.len() >= 3 && chars.take(2).all(|c| c.is_ascii_hexdigit())
```

as we could have any remaining characters still in the string due to calling it with `segment[i..]`. ^^